src/Bicep.Core/TypeSystem/TypeKind.cs

```diff
@@ -48,5 +48,10 @@ public enum TypeKind
         /// Discriminated object type.
         /// </summary>
         DiscriminatedObject,
+
+        /// <summary>
+        /// Unassignable type
+        /// </summary>
+        Unassignable,
```

Comment:
How does this differ from `Never` btw? Should we combine the two?

Reply:
Removed this one and using "never"